pandas/core/window.py

```diff
@@ -1046,16 +1071,21 @@ def validate(self):
 
             # this will raise ValueError on non-fixed freqs
             self.window = freq.nanos
+            if self.closed == 'both':
+                self.window += 1
             self.win_type = 'freq'
 
             # min_periods must be an integer
             if self.min_periods is None:
                 self.min_periods = 1
-
-        elif not is_integer(self.window):
-            raise ValueError("window must be an integer")
-        elif self.window < 0:
-            raise ValueError("window must be non-negative")
+        else:
```

Comment:
why are you changing this? this is an independent check and must be done that way